File: hr.python.answer/laborerIo.py
import menu
import console
from laborer import Laborer

class LaborerIo:
    def __init__(self, laborerDao):
        self.laborerDao = laborerDao
        menuTmp = ''
        for key, val in menu.menu.items():
            menuTmp += f'{key}.{val}    '
        self.menu = menuTmp
    
    def choose(self):
        choice = 0

        while True:
            choice = console.inNum(self.menu)
            if choice >= len(menu.menu):
                console.err('없는 메뉴 번호입니다.')
            else:
                break
        return choice
    
    def play(self):
        while True:
            choice = self.choose()
            if choice:
                if choice == 1: self.listLaborers()
                elif choice == 2: self.addLaborer()
                elif choice == 3: self.fixLaborer()
                elif choice == 4: self.delLaborer()
            else:
                break
    
    def listLaborers(self):
        laborers = self.laborerDao.selectLaborers()

        print('ID   이름    입사일')
        print('--------------------')

        if laborers:
            for laborer in laborers:
                print(laborer)
        else: print('노동자가 없습니다.')
    
    def addLaborer(self):
        laborerName = console.inStr('노동자명을 입력하세요.', 5)

        if laborerName != '0':
            hireDate = console.inDate('입사일을 입력하세요.')
            self.laborerDao.insertLaborer(laborerName, hireDate)
            print('노동자를 추가했습니다.')
        else: print('추가 취소합니다.')
    
    def fixLaborer(self):
        laborerId = console.inNum('노동자 ID를 입력하세요.')

        if laborerId:
            try:
                laborerName = console.inStr('노동자명을 입력하세요.', 5)
                hireDate = console.inDate('입사일을 입력하세요.')
                self.laborerDao.updateLaborer(Laborer(laborerId, laborerName, hireDate))
                print('노동자를 수정했습니다.')
            except LookupError: console.err('해당 노동자가 없습니다.')
        else: print('추가 취소합니다.')

    def delLaborer(self):
        laborerId = console.inNum('노동자 ID를 입력하세요.')

        if laborerId:
            try:
                self.laborerDao.deleteLaborer(laborerId)
                print('노동자를 삭제했습니다.')
            except LookupError: console.err('해당 노동자가 없습니다.')
        else: print('삭제 취소합니다.')

File: 김다훈.hr.java/src/com/my/hr/io/LaborerIo.java
package com.my.hr.io;

import java.util.List;

import com.my.hr.domain.Laborer;
import com.my.hr.service.LaborerService;

public class LaborerIo {
	private LaborerService laborerService;
	
	public LaborerIo(LaborerService laborerService) {
		this.laborerService = laborerService;
	}
	
	public void play() {
		Boolean run = true;
		int chooseNum = 0;
		int laborerId = 0;
		String laborerName = "";
		String startDate = "";
		
		
		
		while(run) {
			Console.info("주메뉴 : 1.목록  2.추가  3.수정  4.삭제  0.종료\n");
			chooseNum = Console.inNum("메뉴번호를 선택하세요 : ");
			
//			laborerId = Console.indexNum("노동자ID: ");
//			laborerName = Console.inStr("노동자명: ");
//			startDate = Console.inDate("입사일: ");
			
			switch(chooseNum) {
				case 1 -> {
					List<Laborer> laborers = laborerService.getLaborerList();
					if(laborers.size() == 0) Console.info("노동자가 없습니다.\n");
					else {
						Console.info("|노동자ID|노동자명|입사일|");
						for(Laborer laborer: laborers) Console.info(laborer);
					}				
				}
				case 2 -> {
					laborerName = Console.inStr("노동자명: ");
					if(laborerName.equals("0")) {
						Console.info("앱을 종료합니다.");
						run = false;
						break;
					}
					startDate = Console.inDate("입사일: ");
					laborerService.addLaborer(new Laborer(laborerName, startDate));
				}
				case 3 -> {
					laborerId = Console.indexNum("변경할 노동자ID: ");
					if(laborerId == 0) {
						Console.info("앱을 종료합니다.");
						run = false;
						break;
					}
					laborerName = Console.inStr("변경할 노동자명: ");
					startDate = Console.inDate("변경할 입사일: ");
					laborerService.fixLaborer(laborerId, laborerName, startDate);
					Console.info(laborerId + "번" + "노동자데이터를 수정하였습니다.");
				}
				case 4 -> {
					laborerId = Console.indexNum("삭제할 노동자ID: ");
					if(laborerId == 0) {
						Console.info("앱을 종료합니다.");
						run = false;
						break;
					}
					laborerService.delLaborer(laborerId);
					Console.info(laborerId + "번" + "노동자데이터를 삭제하였습니다.");
				}
				default -> {
					Console.info("앱을 종료합니다.");
					run = false;
				}
			}
		}
	}
}
